drift/lib/src/sentry_span_helper.dart

```diff
@@ -9,16 +9,17 @@ class SentrySpanHelper {
   final String _origin;
   ISentrySpan? _parentSpan;
 
-  SentrySpanHelper(this._origin, {Hub? hub}) : _hub = hub ?? HubAdapter();
+  SentrySpanHelper(this._origin, {Hub? hub}) : _hub = hub ?? HubAdapter() {
+    _parentSpan = hub?.getSpan();
+  }
 
   Future<T> asyncWrapInSpan<T>(
     String description,
     Future<T> Function() execute, {
     String? dbName,
     String? operation,
   }) async {
-    final currentSpan = _parentSpan ?? _hub.getSpan();
-    final span = currentSpan?.startChild(
+    final span = _parentSpan?.startChild(
```

Comment:
Wouldn't we have the issue that no span were created if there is no parent span? In the case that upon initialization, the hub does not have a span.

Reply:
Also won't attach to the new one if the hub span changes when calling this method.

Reply:
that's the behaviour we have for all other integrations as well, they require an active transaction

Reply:
maybe we should check for the hub span when we try to instrument

Reply:
also I think you commented an older commit, I was confused at first because I changed the impl to use a stack